### PR TITLE
fix InputNumber在父页面组件update并re-render的时候，会失去自身之前的数字选中状态

### PR DESCRIPTION
**Pull request by hengkx**
ref #https://github.com/ant-design/ant-design/issues/12808

### CODE REVIEW

**Comment by amourguo** [comment]
Is there any workaround to make it work properly @3.10.2 ?

**Comment by hengkx** [comment]
@amourguo If there is no problem with this PR, merge it into the post. Update `rc-input-number` dependencies

File: src/index.js [hunk 1]
```diff
@@ -533,7 +533,7 @@ export default class InputNumber extends React.Component {
     if (typeof val === 'number') {
       result =
         ((precisionFactor * val + precisionFactor * step * rat) /
-        precisionFactor).toFixed(precision);
+          precisionFactor).toFixed(precision);
```
Comment (afc163) [inline src/index.js]: Don't change this.

Comment (hengkx) [reply src/index.js]: The editor is automatically formatted.
removed.

**Pull request merged**